test/unit/puppetlabs/puppetserver/ruby/http_client_test.clj

```diff
@@ -176,59 +173,72 @@ jruby-config :- jruby-schemas/JRubyConfig
 (deftest test-ruby-http-client
   (jetty9/with-test-webserver ring-app port
     (with-scripting-container sc
-      (with-http-client sc port {:use-ssl false}
-      (testing "HTTP GET"
-        (is (= "hi" (.runScriptlet sc "$c.get('/', {}).body"))))
-      (testing "HTTP POST"
-        (is (= "hi" (.runScriptlet sc "$c.post('/', 'foo', {}).body"))))))))
+      (with-http-client sc {}
+        (let [url (str "http://localhost:" port)]
+          (testing "HTTP GET"
+            (is (= "hi" (.runScriptlet sc (format "$c.get(URI('%s')).body" url)))))
+          (testing "HTTP POST"
+            (is (= "hi" (.runScriptlet sc (format "$c.post(URI('%s'), 'foo').body" url))))))))))
+
+(deftest http-escaped-urls-test
+  (jetty9/with-test-webserver ring-app port
+    (with-scripting-container sc
+      (with-http-client sc {}
+        (let [url (str "http://localhost:" port)]
+          (testing "HTTP GET"
+            (.runScriptlet sc (str "$response = $c.get(URI('" url "/a%20b%3Fc'), params: { foo: 'bar' })"))
+            (is (= 200 (.runScriptlet sc "$response.code")))
+            (is (= (str url "/a%20b%3Fc?foo=bar") (.runScriptlet sc "$response.url.to_s")))))))))
```

Comment:
I think this is one area where puppet/ruby behaves differently. The response always contains the URL that the caller specified, not the URL that results when query parameters are "merged".

Reply:
Okay, then I think we want to also. I'll put this back so that that happens.